SYCL/ESIMD/api/functional/operators/operator_decrement_and_increment.hpp

```diff
@@ -0,0 +1,283 @@
+struct post_increment {
+  static std::string get_description() { return "post increment"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(const DataT *const ref_data,
+                             DataT *const source_simd_out,
+                             DataT *const result_simd_out) {
+    auto source_simd = esimd::simd<DataT, NumElems>();
+    source_simd.copy_from(ref_data);
+    esimd::simd<DataT, NumElems> result_simd = source_simd++;
+    source_simd.copy_to(source_simd_out);
+    result_simd.copy_to(result_simd_out);
+  }
+
+  template <typename DataT> static DataT apply_operator(DataT &val) {
+    return val++;
+  }
+
+  static constexpr bool is_increment() { return true; }
+};
+
+template <typename DataT, int NumElems, typename TestCaseT>
+class IncrementAndDecrementTestDescription : public ITestDescription {
+public:
+  IncrementAndDecrementTestDescription(size_t index, DataT retrieved_val,
+                                       DataT expected_val,
+                                       const std::string &simd_type,
+                                       const std::string &data_type)
+      : m_data_type(data_type), m_retrieved_val(retrieved_val),
+        m_expected_val(expected_val), m_index(index), m_simd_type(simd_type) {}
+
+  std::string to_string() const override {
+    std::string log_msg("Failed for simd<");
+
+    log_msg += m_data_type + ", " + std::to_string(NumElems) + ">";
+    log_msg += ", retrieved: " + std::to_string(m_retrieved_val);
+    log_msg += ", expected: " + std::to_string(m_expected_val);
+    log_msg += ", at index: " + std::to_string(m_index);
+    log_msg += " for simd as a " + m_simd_type;
+    log_msg += " " + TestCaseT::get_description() + " operator.";
+
+    return log_msg;
+  }
+
+private:
+  const std::string m_data_type;
+  const DataT m_retrieved_val;
+  const DataT m_expected_val;
+  const size_t m_index;
+  const std::string m_simd_type;
+};
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename IsAccuracyTestT, typename DataT, typename SizeT,
+          typename TestCaseT>
+class run_test {
+  static constexpr int NumElems = SizeT::value;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+    std::vector<DataT> ref_data;
+
+    if constexpr (!IsAccuracyTestT::value) {
+      ref_data = generate_ref_data<DataT, NumElems>();
+
+      if constexpr (TestCaseT::is_increment()) {
+        mutate(ref_data, mutator::For_addition<DataT>(1));
+      } else if constexpr (!TestCaseT::is_increment()) {
+        mutate(ref_data, mutator::For_subtraction<DataT>(1));
+      }
+    } else {
+      static const DataT min = value<DataT>::lowest();
+      static const DataT denorm_min = value<DataT>::denorm_min();
+      static const DataT max = value<DataT>::max();
+      static const DataT zero_point_one = static_cast<DataT>(0.1);
```

Comment:
```suggestion
      static const DataT unexact = static_cast<DataT>(0.1);
```
Please, avoid naming any variable based on its exact value. Generalize the code

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/827/commits/7ff74964cc8f28e39cce66778060f321038d13cc#diff-957362298136f7a2e0ab66c1cff4239caca52ea84dab6e6b2728ccc9411ebe0eR172)